File: app/src/main/java/com/soundpola/app/ui/drafts/DraftsScreen.kt
package com.soundpola.app.ui.drafts

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.soundpola.app.data.SoundMemory
import com.soundpola.app.data.SoundRepository
import com.soundpola.app.data.SoundStatus
import com.soundpola.app.data.formatDuration
import com.soundpola.app.data.formatRecordedAt
import com.soundpola.app.ui.components.FilterChipRow
import com.soundpola.app.ui.components.PageHeader
import com.soundpola.app.ui.components.PrimaryButton
import com.soundpola.app.ui.components.SoundVisualCanvas
import com.soundpola.app.ui.components.StatusChip
import com.soundpola.app.ui.theme.CanvasBg
import com.soundpola.app.ui.theme.Ink400
import com.soundpola.app.ui.theme.Ink600
import com.soundpola.app.ui.theme.Ink950
import com.soundpola.app.ui.theme.Primary50
import com.soundpola.app.ui.theme.Primary500
import com.soundpola.app.ui.theme.Primary700
import com.soundpola.app.ui.theme.Radii
import com.soundpola.app.ui.theme.Spacing
import com.soundpola.app.ui.theme.White

private val DraftFilters = listOf("全部", "已暂存", "处理中", "失败")

/** first.md §7.5 Drafts 暂存列表 — designstyle §10.3 */
@Composable
fun DraftsScreen(
    onOpenDetail: (String) -> Unit,
    onPress: (String) -> Unit,
    onStartRecord: () -> Unit,
) {
    var filter by remember { mutableStateOf("全部") }
    var playingId by remember { mutableStateOf<String?>(null) }
    val all = SoundRepository.drafts()
    val drafts = when (filter) {
        "已暂存" -> all.filter { it.status == SoundStatus.Drafted }
        "处理中" -> all.filter { it.status == SoundStatus.Writing || it.status == SoundStatus.ChainPending }
        "失败" -> all.filter { it.status == SoundStatus.WriteFailed || it.status == SoundStatus.ChainFailed }
        else -> all
    }

    Column(Modifier.fillMaxSize().background(CanvasBg)) {
        PageHeader(title = "Drafts", subtitle = "${all.size} 条暂存")
        Spacer(Modifier.height(Spacing.tight))
        FilterChipRow(options = DraftFilters, selected = filter, onSelect = { filter = it })
        Spacer(Modifier.height(Spacing.item))

        if (drafts.isEmpty()) {
            Column(
                Modifier.fillMaxSize(),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center,
            ) {
                SoundVisualCanvas(
                    seed = 1,
                    active = false,
                    modifier = Modifier
                        .size(140.dp)
                        .clip(RoundedCornerShape(Radii.collectionCard))
                        .background(Primary50),
                )
                Spacer(Modifier.height(Spacing.section))
                Text("还没有暂存的声音", color = Ink950, fontWeight = FontWeight.SemiBold)
                Spacer(Modifier.height(8.dp))
                Text("去捕捉此刻的声音", color = Ink600, fontSize = 14.sp)
                Spacer(Modifier.height(Spacing.section))
                Box(Modifier.width(200.dp)) {
                    PrimaryButton(text = "开始录音", onClick = onStartRecord)
                }
            }
        } else {
            LazyColumn(
                modifier = Modifier.padding(horizontal = Spacing.pageHorizontal),
                verticalArrangement = Arrangement.spacedBy(Spacing.tight),
            ) {
                items(drafts, key = { it.id }) { item ->
                    DraftCard(
                        item = item,
                        playing = playingId == item.id,
                        onPlay = { playingId = if (playingId == item.id) null else item.id },
                        onPress = { onPress(item.id) },
                        onOpen = { onOpenDetail(item.id) },
                    )
                }
                item { Spacer(Modifier.height(Spacing.section)) }
            }
        }
    }
}

@Composable
private fun DraftCard(
    item: SoundMemory,
    playing: Boolean,
    onPlay: () -> Unit,
    onPress: () -> Unit,
    onOpen: () -> Unit,
) {
    val showProgress = item.status == SoundStatus.Writing || item.status == SoundStatus.ChainPending
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(Radii.card))
            .background(White)
            .clickable(onClick = onOpen)
            .padding(Spacing.tight),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Box(Modifier.size(72.dp).clip(RoundedCornerShape(Radii.input)).background(Primary50)) {
            SoundVisualCanvas(
                seed = item.visualSeed,
                active = playing,
                showProgressRing = showProgress,
                progress = 0.65f,
                modifier = Modifier.fillMaxSize().padding(6.dp),
            )
        }
        Spacer(Modifier.width(Spacing.tight))
        Column(Modifier.weight(1f)) {
            Text(item.title, color = Ink950, fontWeight = FontWeight.SemiBold, fontSize = 16.sp)
            Spacer(Modifier.height(4.dp))
            Text(
                "${formatRecordedAt(item.recordedAtMillis)} · ${formatDuration(item.durationSec)} · ${item.category}",
                color = Ink600,
                fontSize = 12.sp,
            )
            if (item.locationLabel != "地点未记录") {
                Text(item.locationLabel, color = Ink400, fontSize = 12.sp)
            }
            Spacer(Modifier.height(8.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(Spacing.chip), verticalAlignment = Alignment.CenterVertically) {
                StatusChip(item.status)
                if (item.status == SoundStatus.Drafted || item.status == SoundStatus.WriteFailed) {
                    Text(
                        "Press",
                        color = Primary500,
                        fontWeight = FontWeight.SemiBold,
                        fontSize = 13.sp,
                        modifier = Modifier.clickable(onClick = onPress),
                    )
                }
                if (item.status == SoundStatus.ChainFailed) {
                    Text("重试上链", color = Primary700, fontSize = 13.sp, modifier = Modifier.clickable(onClick = onPress))
                }
            }
        }
        Box(
            Modifier.size(36.dp).clip(CircleShape).background(Primary50).clickable(onClick = onPlay),
            contentAlignment = Alignment.Center,
        ) {
            Text(if (playing) "‖" else "▶", color = Ink950, fontSize = 12.sp)
        }
    }
}

@Composable
fun DraftDetailScreen(id: String, onBack: () -> Unit, onPress: () -> Unit, onDeleted: () -> Unit) {
    val item = SoundRepository.get(id)
    var confirmDelete by remember { mutableStateOf(false) }
    var playing by remember { mutableStateOf(false) }

    if (item == null) {
        Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Text("声音不存在", color = Ink600)
        }
    } else {
        Column(
            Modifier.fillMaxSize().background(CanvasBg).padding(Spacing.pageHorizontal),
        ) {
            Row(Modifier.fillMaxWidth().padding(vertical = Spacing.item), horizontalArrangement = Arrangement.SpaceBetween) {
                Text("返回", color = Ink600, modifier = Modifier.clickable(onClick = onBack))
                Text("详情", color = Ink950, fontWeight = FontWeight.SemiBold)
                Text("删除", color = Ink400, modifier = Modifier.clickable { confirmDelete = true })
            }
            Box(
                Modifier
                    .fillMaxWidth()
                    .height(300.dp)
                    .clip(RoundedCornerShape(Radii.collectionCard))
                    .background(Primary50)
                    .clickable { playing = !playing },
                contentAlignment = Alignment.Center,
            ) {
                SoundVisualCanvas(
                    seed = item.visualSeed,
                    active = playing,
                    modifier = Modifier.fillMaxSize().padding(Spacing.section),
                )
            }
            Spacer(Modifier.height(Spacing.section))
            Text(item.title, color = Ink950, fontSize = 24.sp, fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.height(8.dp))
            StatusChip(item.status)
            Spacer(Modifier.height(Spacing.tight))
            Text("#${item.category}", color = Ink600, fontSize = 14.sp)
            if (item.description.isNotBlank()) {
                Spacer(Modifier.height(Spacing.tight))
                Text(item.description, color = Ink600, fontSize = 15.sp, lineHeight = 24.sp)
            }
            Spacer(Modifier.height(Spacing.item))
            Text(
                "${formatRecordedAt(item.recordedAtMillis)} · ${item.locationLabel} · ${formatDuration(item.durationSec)}",
                color = Ink400,
                fontSize = 12.sp,
            )
            Spacer(Modifier.weight(1f))
            PrimaryButton(
                text = if (item.status == SoundStatus.ChainFailed) "重试上链" else "写入声片",
                onClick = onPress,
            )
            Spacer(Modifier.height(Spacing.item))
        }
    }

    if (confirmDelete && item != null) {
        AlertDialog(
            onDismissRequest = { confirmDelete = false },
            title = { Text("删除这段声音？") },
            text = { Text("录音、声音视觉和相关记忆信息将被永久移除。") },
            confirmButton = {
                TextButton(onClick = {
                    if (SoundRepository.delete(id)) onDeleted()
                    confirmDelete = false
                }) { Text("确认删除") }
            },
            dismissButton = { TextButton(onClick = { confirmDelete = false }) { Text("取消") } },
        )
    }
}

File: mobile/lib/screens/drafts/drafts_screen.dart
import 'package:flutter/material.dart';
import 'dart:async';
import '../../data/sound_repository.dart';
import '../../cloud/cloud_media_client.dart';
import '../../services/audio_playback_service.dart';
import '../../services/auth_service.dart';
import '../../services/mint_pipeline.dart';
import '../../services/visual_shape_service.dart';
import '../../theme/app_colors.dart';
import '../../theme/app_dimens.dart';
import '../../widgets/design_components.dart';
import '../../widgets/sound_visual.dart';

const _filters = ['全部', '已暂存', '处理中', '失败'];

class DraftsScreen extends StatefulWidget {
  const DraftsScreen({
    super.key,
    required this.onOpenDetail,
    required this.onMint,
    required this.onChain,
    required this.onPress,
    required this.onStartRecord,
    required this.onLogin,
  });

  final ValueChanged<String> onOpenDetail;
  final ValueChanged<String> onMint;
  final ValueChanged<String> onChain;
  final ValueChanged<String> onPress;
  final VoidCallback onStartRecord;
  final VoidCallback onLogin;

  @override
  State<DraftsScreen> createState() => _DraftsScreenState();
}

class _DraftsScreenState extends State<DraftsScreen> {
  String _filter = '全部';

  List<SoundMemory> _filtered(List<SoundMemory> items) {
    return switch (_filter) {
      '已暂存' => items.where((s) => s.status == SoundStatus.drafted).toList(),
      '处理中' => items
          .where((s) =>
              s.status == SoundStatus.writing ||
              s.status == SoundStatus.chainPending)
          .toList(),
      '失败' => items
          .where((s) =>
              s.status == SoundStatus.writeFailed ||
              s.status == SoundStatus.chainFailed)
          .toList(),
      _ => items,
    };
  }

  @override
  Widget build(BuildContext context) {
    return AnimatedBuilder(
      animation: Listenable.merge([
        SoundRepository.instance,
        AuthService.instance,
        MintPipeline.instance,
      ]),
      builder: (context, _) {
        final items = _filtered(SoundRepository.instance.drafts);
        final loggedIn = AuthService.instance.isLoggedIn;
        return ColoredBox(
          color: AppColors.bgPrimary,
          child: SafeArea(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                PageHeader(
                  title: 'Drafts',
                  subtitle: '${SoundRepository.instance.drafts.length} 段暂存',
                ),
                if (!loggedIn) ...[
                  LoginHintCard(onLogin: widget.onLogin),
                  const SizedBox(height: AppSpacing.item),
                ],
                FilterChipRow(
                  options: _filters,
                  selected: _filter,
                  onSelect: (v) => setState(() => _filter = v),
                ),
                const SizedBox(height: AppSpacing.item),
                Expanded(
                  child: items.isEmpty
                      ? _EmptyDrafts(onStartRecord: widget.onStartRecord)
                      : ListView.separated(
                          padding: const EdgeInsets.symmetric(
                            horizontal: AppSpacing.pageHorizontal,
                            vertical: AppSpacing.tight,
                          ),
                          itemCount: items.length,
                          separatorBuilder: (_, _) =>
                              const SizedBox(height: AppSpacing.tight),
                          itemBuilder: (context, index) {
                            final item = items[index];
                            return _DraftCard(
                              item: item,
                              onOpen: () => widget.onOpenDetail(item.id),
                              onMint: () => widget.onMint(item.id),
                              onChain: () => widget.onChain(item.id),
                              onPress: () => widget.onPress(item.id),
                            );
                          },
                        ),
                ),
              ],
            ),
          ),
        );
      },
    );
  }
}

class _EmptyDrafts extends StatelessWidget {
  const _EmptyDrafts({required this.onStartRecord});
  final VoidCallback onStartRecord;

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Padding(
        padding: const EdgeInsets.all(AppSpacing.pageHorizontal),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            const SizedBox(
              height: 120,
              width: 120,
              child: SoundVisualCanvas(seed: 42, active: false),
            ),
            const SizedBox(height: AppSpacing.item),
            const Text(
              '还没有暂存的声音',
              style: TextStyle(color: AppColors.textPrimary, fontSize: 16),
            ),
            const SizedBox(height: 8),
            const Text(
              '去捕捉此刻的声音',
              style: TextStyle(color: AppColors.textSecondary),
            ),
            const SizedBox(height: AppSpacing.section),
            PrimaryButton(text: '开始捕捉', onPressed: onStartRecord),
          ],
        ),
      ),
    );
  }
}

class _DraftCard extends StatelessWidget {
  const _DraftCard({
    required this.item,
    required this.onOpen,
    required this.onMint,
    required this.onChain,
    required this.onPress,
  });

  final SoundMemory item;
  final VoidCallback onOpen;
  final VoidCallback onMint;
  final VoidCallback onChain;
  final VoidCallback onPress;

  @override
  Widget build(BuildContext context) {
    final actions = _actionsFor(item.status);

    return Material(
      color: AppColors.surface1,
      borderRadius: BorderRadius.circular(AppRadii.card),
      child: InkWell(
        onTap: onOpen,
        borderRadius: BorderRadius.circular(AppRadii.card),
        child: Padding(
          padding: const EdgeInsets.all(AppSpacing.cardPadding),
          child: Column(
            children: [
              AspectRatio(
                aspectRatio: 16 / 10,
                child: ClipRRect(
                  borderRadius: BorderRadius.circular(AppRadii.chip),
                  child: ColoredBox(
                    color: AppColors.surface2,
                    child: SoundVisualCanvas(
                      seed: item.visualSeed,
                      mode: SoundVisualMode.complete,
                      shape: VisualShapeService.instance
                          .peek(item.visualPath),
                    ),
                  ),
                ),
              ),
              const SizedBox(height: AppSpacing.tight),
              Row(
                children: [
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          item.title,
                          maxLines: 1,
                          overflow: TextOverflow.ellipsis,
                          style: const TextStyle(
                            fontWeight: FontWeight.w600,
                            fontSize: 16,
                            color: AppColors.textPrimary,
                          ),
                        ),
                        const SizedBox(height: 4),
                        Text(
                          '${item.category} · ${formatDuration(item.durationSec)} · ${item.locationLabel}',
                          maxLines: 1,
                          overflow: TextOverflow.ellipsis,
                          style: const TextStyle(
                            color: AppColors.textTertiary,
                            fontSize: 12,
                          ),
                        ),
                        const SizedBox(height: 6),
                        StatusChip(status: item.status),
                      ],
                    ),
                  ),
                  for (final action in actions)
                    TextButton(
                      onPressed: action.enabled
                          ? switch (action.type) {
                              _ActionType.cloud => onMint,
                              _ActionType.chain => onChain,
                              _ActionType.press => onPress,
                            }
                          : null,
                      child: Text(
                        action.label,
                        style: TextStyle(
                          color: action.enabled
                              ? AppColors.accent
                              : AppColors.textTertiary,
                        ),
                      ),
                    ),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}

enum _ActionType { cloud, chain, press }

class _DraftAction {
  const _DraftAction(this.label, this.type, {this.enabled = true});
  final String label;
  final _ActionType type;
  final bool enabled;
}

List<_DraftAction> _actionsFor(SoundStatus status) {
  switch (status) {
    case SoundStatus.drafted:
      return [const _DraftAction('上云', _ActionType.cloud)];
    case SoundStatus.writing:
      return [const _DraftAction('处理中', _ActionType.cloud, enabled: false)];
    case SoundStatus.cloudReady:
      return [
        const _DraftAction('写入声片', _ActionType.press),
        const _DraftAction('上链', _ActionType.chain),
      ];
    case SoundStatus.chainPending:
      return [
        const _DraftAction('写入声片', _ActionType.press),
        const _DraftAction('上链中', _ActionType.chain, enabled: false),
      ];
    case SoundStatus.chainReady:
      return [const _DraftAction('写入声片', _ActionType.press)];
    case SoundStatus.writeFailed:
      return [const _DraftAction('重试上云', _ActionType.cloud)];
    case SoundStatus.chainFailed:
      return [
        const _DraftAction('写入声片', _ActionType.press),
        const _DraftAction('重试上链', _ActionType.chain),
      ];
    case SoundStatus.collected:
      return [];
  }
}

class DraftDetailScreen extends StatefulWidget {
  const DraftDetailScreen({
    super.key,
    required this.id,
    required this.onBack,
    required this.onMint,
    required this.onChain,
    required this.onPress,
    required this.onDeleted,
  });

  final String id;
  final VoidCallback onBack;
  final VoidCallback onMint;
  final VoidCallback onChain;
  final VoidCallback onPress;
  final VoidCallback onDeleted;

  @override
  State<DraftDetailScreen> createState() => _DraftDetailScreenState();
}

class _DraftDetailScreenState extends State<DraftDetailScreen> {
  bool _playing = false;
  bool _editing = false;
  final _player = AudioPlaybackService.instance;
  late final TextEditingController _titleCtrl;
  late final TextEditingController _descCtrl;
  String? _category;
  StreamSubscription<void>? _completeSub;

  @override
  void initState() {
    super.initState();
    _completeSub = _player.completionStream.listen((_) {
      if (mounted) setState(() => _playing = false);
    });
    final item = SoundRepository.instance.get(widget.id);
    _titleCtrl = TextEditingController(text: item?.title ?? '');
    _descCtrl = TextEditingController(text: item?.description ?? '');
    _category = item?.category;
    if (item != null) {
      unawaited(VisualShapeService.instance.load(item.visualPath));
    }
  }

  @override
  void dispose() {
    _completeSub?.cancel();
    _player.stop();
    _titleCtrl.dispose();
    _descCtrl.dispose();
    super.dispose();
  }

  Future<void> _togglePlay(String? path) async {
    if (path == null || path.isEmpty) {
      setState(() => _playing = !_playing);
      return;
    }
    if (_playing) {
      await _player.stop();
      setState(() => _playing = false);
    } else {
      await _player.play(path);
      setState(() => _playing = true);
    }
  }

  void _saveEdit() {
    final title = _titleCtrl.text.trim();
    if (title.isEmpty || _category == null) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(content: Text('名称与分类不能为空')),
      );
      return;
    }
    SoundRepository.instance.update(
      widget.id,
      (s) => s.copyWith(
        title: title,
        category: _category,
        description: _descCtrl.text.trim(),
      ),
    );
    setState(() => _editing = false);
    unawaited(_syncLabelToCloud(title));
  }

  Future<void> _syncLabelToCloud(String title) async {
    final item = SoundRepository.instance.get(widget.id);
    final contentId = item?.contentId;
    final token = AuthService.instance.cloudToken;
    if (contentId == null || contentId.isEmpty || token == null || token.isEmpty) {
      return;
    }
    try {
      await CloudMediaClient().patchContent(
        token: token,
        contentId: contentId,
        displayLabel: title,
      );
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('云端名称同步失败：$e')),
        );
      }
    }
  }

  Future<void> _confirmDelete() async {
    final ok = await showDialog<bool>(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('删除这段声音？'),
        content: const Text('录音、声音视觉和相关记忆信息将被永久移除。'),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx, false),
            child: const Text('取消'),
          ),
          TextButton(
            onPressed: () => Navigator.pop(ctx, true),
            child: const Text('确认删除', style: TextStyle(color: AppColors.error)),
          ),
        ],
      ),
    );
    if (ok == true) {
      if (SoundRepository.instance.delete(widget.id)) {
        widget.onDeleted();
      } else if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(content: Text('NFC 已写入的声音不可删除，请重试上链')),
        );
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    return AnimatedBuilder(
      animation: Listenable.merge([
        SoundRepository.instance,
        MintPipeline.instance,
      ]),
      builder: (context, _) {
        final item = SoundRepository.instance.get(widget.id);
        if (item == null) {
          return Scaffold(
            appBar: AppBar(title: const Text('Drafts')),
            body: const Center(child: Text('声音不存在')),
          );
        }

        return Scaffold(
          backgroundColor: AppColors.bgPrimary,
          appBar: AppBar(
            title: const Text('暂存详情'),
            leading: IconButton(
              icon: const Icon(Icons.arrow_back),
              onPressed: widget.onBack,
            ),
            actions: [
              TextButton(
                onPressed: () {
                  if (_editing) {
                    _saveEdit();
                  } else {
                    setState(() => _editing = true);
                  }
                },
                child: Text(
                  _editing ? '保存' : '编辑',
                  style: const TextStyle(color: AppColors.accent),
                ),
              ),
            ],
          ),
          body: SafeArea(
            child: ListView(
              padding: const EdgeInsets.symmetric(
                horizontal: AppSpacing.pageHorizontal,
              ),
              children: [
                GestureDetector(
                  onTap: () => _togglePlay(item.audioPath),
                  child: ClipRRect(
                    borderRadius:
                        BorderRadius.circular(AppRadii.collectionCard),
                    child: AspectRatio(
                      aspectRatio: 1,
                      child: ColoredBox(
                        color: AppColors.surface1,
                        child: Stack(
                          alignment: Alignment.center,
                          children: [
                            SoundVisualCanvas(
                              seed: item.visualSeed,
                              mode: _playing
                                  ? SoundVisualMode.playback
                                  : SoundVisualMode.complete,
                              shape: VisualShapeService.instance
                                  .peek(item.visualPath),
                            ),
                            Icon(
                              _playing
                                  ? Icons.pause_circle_outline
                                  : Icons.play_circle_outline,
                              size: 44,
                              color: AppColors.accent.withValues(alpha: 0.85),
                            ),
                          ],
                        ),
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: AppSpacing.item),
                if (_editing) ...[
                  SpTextField(
                    controller: _titleCtrl,
                    label: '声音名称',
                    maxLength: 20,
                  ),
                  const SizedBox(height: AppSpacing.item),
                  const SectionLabel('分类'),
                  Wrap(
                    spacing: 8,
                    runSpacing: 8,
                    children: soundCategories.map((c) {
                      final active = c == _category;
                      return GestureDetector(
                        onTap: () => setState(() => _category = c),
                        child: Container(
                          padding: const EdgeInsets.symmetric(
                            horizontal: 12,
                            vertical: 8,
                          ),
                          decoration: BoxDecoration(
                            color: active
                                ? AppColors.accent.withValues(alpha: 0.15)
                                : AppColors.surface1,
                            borderRadius: BorderRadius.circular(50),
                            border: Border.all(
                              color: active
                                  ? AppColors.accent
                                  : AppColors.border,
                            ),
                          ),
                          child: Text(
                            c,
                            style: TextStyle(
                              color: active
                                  ? AppColors.accent
                                  : AppColors.textTertiary,
                            ),
                          ),
                        ),
                      );
                    }).toList(),
                  ),
                  const SizedBox(height: AppSpacing.item),
                  SpTextField(
                    controller: _descCtrl,
                    label: '描述（选填）',
                    maxLines: 3,
                    maxLength: 200,
                  ),
                ] else ...[
                  Text(
                    item.title,
                    style: const TextStyle(
                      fontSize: 24,
                      fontWeight: FontWeight.w600,
                      color: AppColors.textPrimary,
                    ),
                  ),
                  const SizedBox(height: 8),
                  Text(
                    '#${item.category} · ${formatDuration(item.durationSec)}',
                    style: const TextStyle(color: AppColors.textSecondary),
                  ),
                  const SizedBox(height: 8),
                  StatusChip(status: item.status),
                  if (item.description.isNotEmpty) ...[
                    const SizedBox(height: AppSpacing.item),
                    Text(
                      item.description,
                      style: const TextStyle(
                        color: AppColors.textSecondary,
                        height: 1.5,
                      ),
                    ),
                  ],
                  const SizedBox(height: AppSpacing.item),
                  MetaRow(
                    label: '录制时间',
                    value: formatRecordedAt(item.recordedAt),
                  ),
                  MetaRow(label: '地点', value: item.locationLabel),
                  MetaRow(
                    label: '时长',
                    value: formatDuration(item.durationSec),
                  ),
                  MetaRow(label: '设备', value: item.deviceLabel),
                ],
                const SizedBox(height: AppSpacing.block),
                for (final action in _actionsFor(item.status))
                  Padding(
                    padding: const EdgeInsets.only(bottom: AppSpacing.tight),
                    child: PrimaryButton(
                      text: action.label,
                      enabled: action.enabled,
                      onPressed: action.enabled
                          ? switch (action.type) {
                              _ActionType.cloud => widget.onMint,
                              _ActionType.chain => widget.onChain,
                              _ActionType.press => widget.onPress,
                            }
                          : null,
                    ),
                  ),
                const SizedBox(height: AppSpacing.tight),
                SecondaryButton(
                  text: '删除声音',
                  danger: true,
                  onPressed: _confirmDelete,
                ),
                const SizedBox(height: AppSpacing.section),
              ],
            ),
          ),
        );
      },
    );
  }
}
